test/tools/yulopti.cpp

```diff
@@ -171,28 +208,118 @@ class YulOpti
 		}
 	}
 
+	void objectApply(shared_ptr<Object> _object, function<void(Object&)> _fn)
+	{
+		for (auto const& subObjectNode: _object->subObjects) {
+			auto subObject = dynamic_pointer_cast<Object>(subObjectNode);
+
+			if (subObject != nullptr)
+				objectApply(subObject, _fn);
+		}
+
+		_fn(*_object);
+	}
+
+	void analyze(ErrorReporter& errorReporter)
+	{
+		objectApply(m_object, [&](Object& object) {
```

Comment:
Maybe it is just me, but I think that the definition of the lambda expression here makes it a little bit hard to read, since it is fairly long. If you think it is a good idea, we could assign it to a local var before and then pass it to `objectApply` so it is more readable.
Other option, since there are breaking lines, could be to break at opening parenthesis of `objectApply`:
```suggestion
		objectApply(
			m_object,
			[&](Object& _object)
			{
```
And do we need `[&]` ? Nothing is being captured from outer scope ...
Also, coding style (underscore for param var name).

Reply:
Went with the line split since the local var would've required some kind of name that'd be a duplicate of something.